lib/config/migrations/custom/dep-types-migration.ts

```diff
@@ -0,0 +1,36 @@
+import is from '@sindresorhus/is';
+import type { PackageRule, PackageRuleInputConfig } from '../../types';
+import { AbstractMigration } from '../base/abstract-migration';
+
+interface DepTypesRule extends PackageRule, PackageRuleInputConfig {}
+export class DepTypesMigration extends AbstractMigration {
+  override readonly deprecated = true;
+  override readonly propertyName = /.*[d|D]ependencies$|engines$|depTypes$/;
```

Comment:
```suggestion
  override readonly propertyName = /^(?:(?:d|devD|optionalD|peerD)ependencies|engines|depTypes)$/;
```

We should be more defense like before here.